lib/orderbook/MatchingEngine.ts

```diff
@@ -149,6 +151,15 @@ class MatchingEngine {
 
     return matchingResult;
   }
+
+  public removeOwnOrder = (orderId: string): orders.StampedOwnOrder | null => {
+    return this.priorityQueues.buyOrders.removeOne(order => order.id === orderId) ||
```

Comment:
For performance reasons I'd guess it's faster to check whether the order is a buy or sell (check if quantity is positive) and then do removeOne on either `buyOrders` or `sellOrders`, but not both. Currently it would be possible for `removeOne` to execute twice if it's not a buy order, which means we iterate through all orders once more than necessary.

Reply:
This information (`quantity`) isn't available in the current design.
It can be added as a parameter to the gRPC/CLI call, but I don't think it's a good idea. 
It's also possible to do a fast lookup on the separated lists on `OrderBook`, but this area is subject to change, so I don't want to introduce more stuff there. And in it's current design, the lists are not unified, so I need to check on both buy/sell lists to find the order. But yes, it will still be faster, and then the complete object will be available.

Is that what you meant when you suggested to accept `StampedOrder` as a parameter? that we'll retrieve it from the separated lists by `orderId`, and pass it to this function?